dbuf-core/src/parser/lexer.rs

```diff
@@ -0,0 +1,87 @@
+use chumsky::span::SimpleSpan;
+use logos::Logos;
+use unescape::unescape;
+
+#[derive(Logos, Debug, PartialEq, Clone)]
+pub enum Token {
+    #[token("message")]
+    Message,
+    #[token("enum")]
+    Enum,
+
+    #[token("true", |_| true)]
+    #[token("false", |_| false)]
+    BoolLiteral(bool),
+    #[regex(r"[0-9]+", |lex| lex.slice().parse().ok())]
+    IntLiteral(i64),
+    #[regex(r"[0-9]+u", |lex| lex.slice()[..lex.slice().len()-1].parse().ok())]
```

Comment:
This slicing looks non-obvious so I'd extract it into a separate function with a comment explaining why `..lex.slice().len()-1` is needed